src/main/java/org/elasticsearch/indices/recovery/RecoveryTarget.java

```diff
@@ -398,6 +401,14 @@ public void messageReceived(RecoveryCleanFilesRequest request, TransportChannel
                 Store.MetadataSnapshot sourceMetaData = request.sourceMetaSnapshot();
                 try {
                     store.cleanupAndVerify("recovery CleanFilesRequestHandler", sourceMetaData);
+                } catch (CorruptIndexException | IndexFormatTooNewException | IndexFormatTooOldException ex) {
+                    // this is a fatal exception at this stage.
+                    // this means we transferred files from the remote that have not be checksummed and they are
+                    // broken. We have to clean up this shard entirely, remove all files and bubble it up to the
+                    // source shard since this index might be broken there as well? The Source can handle this and checks
+                    // its content on disk if possible.
+                    store.deleteContent(); // clean up and delete all files
```

Comment:
For my understanding here - what is different here from other places where we detection corruption and leave a corruption marker (rather then deleting everything)?